### PR TITLE
remove instances of to_unsigned_integer

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp [hunk 1]
```diff
@@ -1066,12 +1066,11 @@ codet java_bytecode_convert_methodt::convert_instructions(
     {
       PRECONDITION(!i_it->args.empty());
 
-      unsigned target;
-      bool ret=to_unsigned_integer(to_constant_expr(i_it->args[0]), target);
-      DATA_INVARIANT(!ret, "target expected to be unsigned integer");
-      targets.insert(target);
+      auto target=numeric_cast<unsigned>(to_constant_expr(i_it->args[0]));
+      DATA_INVARIANT(target.has_value(), "target expected to be unsigned integer");
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/java_bytecode_convert_method.cpp]: Just use `numeric_cast_v` instead of this convert+check step.

**Review by tautschnig — CHANGES_REQUESTED** [review]
Before reviewing any further and copying the same comment several times: this can be simplified by using `numeric_cast_v`

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: d9428eb151caa89f9b3a1fb11348f8afe4f0b671).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86369065

**Review by romainbrenguier — APPROVED** [review]

File: src/solvers/refinement/string_constraint_generator_format.cpp [hunk 2]
```diff
@@ -486,9 +486,7 @@ utf16_constant_array_to_java(const array_exprt &arr, std::size_t length)
   for(std::size_t i=0; i<arr.operands().size() && i<length; i++)
   {
     PRECONDITION(arr.operands()[i].id()==ID_constant);
-    auto c = numeric_cast<unsigned>(to_constant_expr(arr.operands()[i]));
-    INVARIANT(c.has_value(), "constant should be convertible to unsigned");
-    out[i] = c.value();
+    out[i] = numeric_cast_v<unsigned>(to_constant_expr(arr.operands()[i]));
```
Comment (tautschnig) [inline src/solvers/refinement/string_constraint_generator_format.cpp:491]: The above `PRECONDITION` is rather redundant.

**Review by tautschnig — APPROVED** [review]
Approving to move this forward, but I'd appreciate if the below comment could be considered.

File: src/solvers/refinement/string_constraint_generator_format.cpp (hunk 2 above)
Comment (kroening) [reply src/solvers/refinement/string_constraint_generator_format.cpp:491]: I do like it as it clarifies the contract here well; will move up.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 76e8c1ff0c5a52bd5c340e0c40c32fd57f549578).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87886965

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 790362381439c76c63b070970acb2ee21cdd8e70).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87892779

File: src/solvers/refinement/string_constraint_generator_format.cpp (hunk 2 above)
Comment (martin-cs) [reply src/solvers/refinement/string_constraint_generator_format.cpp:491]: Having the two loops, esp. with different iteration control does seem a bit unnecessary.

File: src/solvers/refinement/string_constraint_generator_format.cpp [hunk 3]
```diff
@@ -482,16 +482,14 @@ std::pair<exprt, string_constraintst> add_axioms_for_format(
 std::string
 utf16_constant_array_to_java(const array_exprt &arr, std::size_t length)
 {
+  for(const auto &op : arr.operands())
+    PRECONDITION(op.id()==ID_constant);
```
Comment (peterschrammel) [inline src/solvers/refinement/string_constraint_generator_format.cpp]: clang-format

**Review by peterschrammel — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 790362381439c76c63b070970acb2ee21cdd8e70).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88668373

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 0209196cc4fbc1c9590e3cc59370f3006e07b302).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88690519

**Pull request merged**